### ISSUE TITLE
[no-if]: Works with arrow function, fails in standard function

### ISSUE THREAD

**Issue by dtom90** [jest-community/eslint-plugin-jest#670]
When `jest/no-if` is enabled, An if statement within `() => {` is caught no problem. But when it's inside `function () {`, it is not caught.

Example:

```js
const condition = true

test('arrow function with condition', () => {
  let sum = 0
  if (condition) { // this fails jest/no-if
    sum = 1 + 2
  }
  expect(sum).toBe(3);
});

test('standard function with condition', function () {
  let sum = 0
  if (condition) { // this is not caught by jest/no-if
    sum = 1 + 2
  }
  expect(sum).toBe(3);
});
```

[Reproducible Code](https://github.com/dtom90/jest-no-if)
